### DISCUSSION TITLE
How to set env variables in the Operator

### DISCUSSION THREAD

**Discussion by githubing** [Apicurio/apicurio-registry#2734] [Q&A] [answered]
Hi, I am testing the Apicurio Registry Operator and it works very well!
Because I want Apicurio manage the roles internally I need to set an env variable, does anybody how to do that? thanks!

**Comment by jsenko** [comment, marked as answer]
Hi @githubing Currently, you have to set any custom variables directly in the Deployment that the Operator created. The Operator won't overwite them. We are aware this is not a good solution, so we're going to add support for setting env. variables directly into the ApicurioRegistry CR in the next release.

**Reply by githubing** [reply]
thanks @jsenko!
I hope you can help me with these last few questions too!
1. is there any way I can set these vars when I deploy the ApicurioRegistry CR at the moment or that has to be inevitably after by modifying the deployment?
2. do you know when roughly the next stable release will be released with this functionality?
3. I need also to set the pods Resources/Limits, how can I do that with current release?

**Reply by jsenko** [reply]
1. It has to be done afterwards :(
2. The next release is planned in September
3. This is one of the open issues as well, I'll try to prioritize it and get it to the next release as well

**Reply by githubing** [reply]
Hi @jsenko, do you know if this functionality has been implemented yet or if any updated ETA?

**Reply by jsenko** [reply]
hi, the env. variable feature has been implemented and is available in the Service Registry operator. Apicurio Registry operator needs a release to the Operator Hub, which is on my TODO list.

**Reply by jsenko** [reply]
As for Resources & Limits we plan to add the whole podTemplate to the CR as an experimental change, since this would solve a whole lot of issues/requests with one feature.

**Reply by githubing** [reply]
do you know when we would be expecting them to be released? thanks
